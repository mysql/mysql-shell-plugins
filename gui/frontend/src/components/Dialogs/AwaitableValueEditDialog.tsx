/*
 * Copyright (c) 2023, 2024, Oracle and/or its affiliates.
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License, version 2.0,
 * as published by the Free Software Foundation.
 *
 * This program is designed to work with certain software (including
 * but not limited to OpenSSL) that is licensed under separate terms, as
 * designated in a particular file or component or in included license
 * documentation.  The authors of MySQL hereby grant you an additional
 * permission to link the program and your derivative works with the
 * separately licensed software that they have either included with
 * the program or referenced in the documentation.
 *
 * This program is distributed in the hope that it will be useful,  but
 * WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See
 * the GNU General Public License, version 2.0, for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program; if not, write to the Free Software Foundation, Inc.,
 * 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA
 */

import { ComponentChild, createRef } from "preact";

import { DialogResponseClosure, IDialogRequest, IDictionary } from "../../app-logic/general-types.js";
import { Semaphore } from "../../supplement/Semaphore.js";
import { ComponentBase } from "../ui/Component/ComponentBase.js";
import { IDialogValidations, IDialogValues, IValueEditDialogShowOptions, ValueEditDialog } from "./ValueEditDialog.js";

export interface IAwaitableDialogResult {
    values: IDialogValues;
    closure: DialogResponseClosure;
}

/**
 * A dialog implementation that provides the necessary tools to allow derived dialogs to return their results to a
 * waiting customer directly. This allows to chain multiple dialogs together.
 */
export class AwaitableValueEditDialog extends ComponentBase {
    #dialogRef = createRef<ValueEditDialog>();
    #signal?: Semaphore<IAwaitableDialogResult>;

    /** @returns a unique ID for this dialog instance. */
    protected get id(): string {
        return "";
    }

    public get dialog(): ValueEditDialog | null {
        return this.#dialogRef.current;
    }

    public render(): ComponentChild {
        return (
            <ValueEditDialog
                ref={this.#dialogRef}
                id={this.id}
                onClose={this.handleCloseDialog}
                onValidate={this.validateInput}
            />
        );
    }

    /**
     * Will be overwritten by descendants.
     *
     * @param request The request to show the dialog.
     *
     * @returns A promise resolving to DialogResponseClosure.Cancel.
     */
    // eslint-disable-next-line @typescript-eslint/no-unused-vars
    public show(request: IDialogRequest): Promise<IDictionary | DialogResponseClosure> {
        return Promise.resolve(DialogResponseClosure.Cancel);
    }

    protected async doShow(valueFactory: () => IDialogValues | undefined,
        options?: IValueEditDialogShowOptions): Promise<IAwaitableDialogResult> {
        this.#signal = new Semaphore();
        this.#dialogRef.current?.show(valueFactory(), options);

        const result = await this.#signal.wait();
        this.#signal = undefined;

        return result;
    }

    /**
     * Will be overwritten by descendants.
     *
     * @param closing Indicates whether the dialog is closing or not.
     * @param values The current values of the dialog.
     *
     * @returns A dictionary of validation messages.
     */
    // eslint-disable-next-line @typescript-eslint/no-unused-vars
    protected validateInput = (closing: boolean, values: IDialogValues): IDialogValidations => {
        return { messages: {} };
    };

    private handleCloseDialog = (closure: DialogResponseClosure, values: IDialogValues): void => {
        this.#signal?.notify({ closure, values });
    };
}
