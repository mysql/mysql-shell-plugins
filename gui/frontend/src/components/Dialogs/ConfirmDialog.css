/*
 * Copyright (c) 2021, 2022, Oracle and/or its affiliates.
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License, version 2.0,
 * as published by the Free Software Foundation.
 *
 * This program is also distributed with certain software (including
 * but not limited to OpenSSL) that is licensed under separate terms, as
 * designated in a particular file or component or in included license
 * documentation.  The authors of MySQL hereby grant you an additional
 * permission to link the program and your derivative works with the
 * separately licensed software that they have included with MySQL.
 * This program is distributed in the hope that it will be useful,  but
 * WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See
 * the GNU General Public License, version 2.0, for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program; if not, write to the Free Software Foundation, Inc.,
 * 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA
 */

 .msg.confirmDialog {
    padding: 0;
    color: var(--foreground);
    min-height: 50px;
}

.msg.confirmDialog #dialogHeading {
    font-size: 1.2em;
    margin-bottom: 20px;
}

.msg.confirmDialog .content {
    white-space: inherit;
}

.msg.confirmDialog .description {
    margin-bottom: 20px;
    font-size: 0.9em;
    color: var(--text-dimmedForeground);
}

.msg.confirmDialog .footer {
    padding: 8px 0
}

.msg.confirmDialog #dialogMessage {
    margin-bottom: 8px;
}
