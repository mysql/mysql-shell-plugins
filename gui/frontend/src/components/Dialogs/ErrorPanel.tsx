/*
 * Copyright (c) 2021, 2023, Oracle and/or its affiliates.
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License, version 2.0,
 * as published by the Free Software Foundation.
 *
 * This program is also distributed with certain software (including
 * but not limited to OpenSSL) that is licensed under separate terms, as
 * designated in a particular file or component or in included license
 * documentation.  The authors of MySQL hereby grant you an additional
 * permission to link the program and your derivative works with the
 * separately licensed software that they have included with MySQL.
 * This program is distributed in the hope that it will be useful,  but
 * WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See
 * the GNU General Public License, version 2.0, for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program; if not, write to the Free Software Foundation, Inc.,
 * 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA
 */

import "./ErrorPanel.css";

import { ComponentChild, createRef } from "preact";

import { requisitions } from "../../supplement/Requisitions";
import { Codicon } from "../ui/Codicon";
import { IComponentState, ComponentBase } from "../ui/Component/ComponentBase";
import { Container, Orientation } from "../ui/Container/Container";
import { Dialog } from "../ui/Dialog/Dialog";
import { Icon } from "../ui/Icon/Icon";
import { Label } from "../ui/Label/Label";
import { Button } from "../ui/Button/Button";

interface IErrorPanelState extends IComponentState {
    caption: string;
    message: string;
}

export class ErrorPanel extends ComponentBase<{}, IErrorPanelState> {

    private dialogRef = createRef<Dialog>();

    public constructor(props: {}) {
        super(props);

        this.state = {
            caption: "",
            message: "",
        };
    }

    public componentDidMount(): void {
        requisitions.register("showError", this.show);
    }

    public componentWillUnmount(): void {
        requisitions.unregister("showError", this.show);
    }

    public render(): ComponentChild {
        const { caption, message } = this.state;

        const className = this.getEffectiveClassNames(["errorPanel"]);

        return (
            <Dialog
                ref={this.dialogRef}
                className={className}
                caption={
                    <>
                        <Icon src={Codicon.Error} />
                        <Label>{caption}</Label>
                    </>
                }
                content={
                    <Container orientation={Orientation.TopDown}>
                        {message && <Label
                            id="errorMessage"
                            caption={message}
                            style={{ whiteSpace: "pre-line" }}
                        />}
                    </Container>
                }
                actions={{
                    end: [
                        <Button
                            key="close"
                            caption="Close"
                            onClick={this.closePanel}
                        />,
                    ],
                }}
            >
            </Dialog>
        );
    }

    private show = (values: string[]): Promise<boolean> => {
        return new Promise((resolve) => {
            if (values.length > 0) {
                let caption = "Error";
                let message = values[0];

                if (values.length > 1) {
                    caption = message;
                    message = values.slice(1).join("\n");
                }

                this.setState({ caption, message }, () => {
                    this.dialogRef.current?.open();
                    resolve(true);
                });
            } else {
                resolve(true);
            }
        });
    };

    private closePanel = (): void => {
        this.dialogRef.current?.close(false);
    };

}
