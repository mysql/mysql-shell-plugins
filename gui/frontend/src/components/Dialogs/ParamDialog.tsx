/*
 * Copyright (c) 2021, 2023, Oracle and/or its affiliates.
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License, version 2.0,
 * as published by the Free Software Foundation.
 *
 * This program is also distributed with certain software (including
 * but not limited to OpenSSL) that is licensed under separate terms, as
 * designated in a particular file or component or in included license
 * documentation.  The authors of MySQL hereby grant you an additional
 * permission to link the program and your derivative works with the
 * separately licensed software that they have included with MySQL.
 * This program is distributed in the hope that it will be useful,  but
 * WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See
 * the GNU General Public License, version 2.0, for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program; if not, write to the Free Software Foundation, Inc.,
 * 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA
 */

import paramIcon from "../../assets/images/parameters.svg";
import "./ParamDialog.css";

import { ComponentChild, createRef } from "preact";
import { Button, IButtonProperties } from "../ui/Button/Button";
import { Codicon } from "../ui/Codicon";
import { IComponentProperties, IComponentState, ComponentBase } from "../ui/Component/ComponentBase";
import { ContentAlignment } from "../ui/Container/Container";
import { Dialog } from "../ui/Dialog/Dialog";
import { Grid } from "../ui/Grid/Grid";
import { GridCell } from "../ui/Grid/GridCell";
import { Icon } from "../ui/Icon/Icon";
import { Input, IInputChangeProperties } from "../ui/Input/Input";
import { Label } from "../ui/Label/Label";

interface IParamDialogProperties extends IComponentProperties {
    caption?: string;
    onClose?: (accepted: boolean, payload?: unknown) => void;
}

interface IParamDialogState extends IComponentState {
    name: string;
    value: string;
}

export class ParamDialog extends ComponentBase<IParamDialogProperties, IParamDialogState> {

    private dialogRef = createRef<Dialog>();

    public constructor(props: IParamDialogProperties) {
        super(props);

        this.state = {
            name: "",
            value: "",
        };

        this.addHandledProperties("caption", "onClose");
    }

    public show = (): void => {
        return this.dialogRef.current?.open();
    };

    public render(): ComponentChild {
        const { caption } = this.props;
        const { name, value } = this.state;

        const className = this.getEffectiveClassNames(["paramDialog"]);
        const dlgCaption = caption ?? "Please enter name/vale pair:";

        return <Dialog
            ref={this.dialogRef}
            className={className}
            onClose={this.closeDialog}
            caption={
                <>
                    <Icon src={Codicon.PassFilled} />
                    <Label>Parameters</Label>
                </>
            }
            content={
                <Grid columns={["128px", "auto", "auto"]} columnGap={8}>
                    <GridCell rowSpan={4}><Icon src={paramIcon} width={128} height={128} /></GridCell>
                    <GridCell columnSpan={2} crossAlignment={ContentAlignment.Center}>
                        <Label id="caption" caption={dlgCaption} />
                    </GridCell>
                    <GridCell className="left" crossAlignment={ContentAlignment.Center}>Parameter name:</GridCell>
                    <GridCell className="right" crossAlignment={ContentAlignment.Center}>
                        <Input id="name" value={name} onChange={this.handleNameChange} />
                    </GridCell>
                    <GridCell className="left" crossAlignment={ContentAlignment.Center}>Parameter value:</GridCell>
                    <GridCell className="right" crossAlignment={ContentAlignment.Center}>
                        <Input id="value" value={value} onChange={this.handleValueChange} />
                    </GridCell>
                </Grid>
            }
            actions={{
                end: [
                    <Button
                        id="ok"
                        key="ok"
                        caption="OK"
                        onClick={this.handleButtonClick}
                    />,
                    <Button
                        id="cancel"
                        key="cancel"
                        caption="Cancel"
                        onClick={this.handleButtonClick}
                    />,
                ],
            }}
        />;
    }

    private handleNameChange = (e: InputEvent, props: IInputChangeProperties): void => {
        this.setState({ name: props.value });
    };

    private handleValueChange = (e: InputEvent, props: IInputChangeProperties): void => {
        this.setState({ value: props.value });
    };

    private handleButtonClick = (e: MouseEvent | KeyboardEvent, props: IButtonProperties): void => {
        this.dialogRef.current?.close(props.id !== "ok");
    };

    private closeDialog = (cancelled: boolean): void => {
        const { onClose } = this.props;
        const { name, value } = this.state;

        onClose?.(cancelled, { name, value });
    };
}
