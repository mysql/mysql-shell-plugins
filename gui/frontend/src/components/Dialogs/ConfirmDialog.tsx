/*
 * Copyright (c) 2021, 2022, Oracle and/or its affiliates.
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License, version 2.0,
 * as published by the Free Software Foundation.
 *
 * This program is also distributed with certain software (including
 * but not limited to OpenSSL) that is licensed under separate terms, as
 * designated in a particular file or component or in included license
 * documentation.  The authors of MySQL hereby grant you an additional
 * permission to link the program and your derivative works with the
 * separately licensed software that they have included with MySQL.
 * This program is distributed in the hope that it will be useful,  but
 * WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See
 * the GNU General Public License, version 2.0, for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program; if not, write to the Free Software Foundation, Inc.,
 * 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA
 */

import "./ConfirmDialog.css";

import React from "react";

import {
    Button, Codicon, Component, Container, Dialog, IComponentProperties, IComponentState, Icon, Label, Orientation,
} from "../ui";
import { DialogResponseClosure, IDictionary } from "../../app-logic/Types";

export interface IConfirmDialogButtons {
    accept?: string;      // Default: "Yes".
    refuse?: string;      // Default: "No".
    alternative?: string; // Default: nothing.
    default?: string;     // Default: nothing.
}

export interface IConfirmDialogProperties extends IComponentProperties {
    onClose?: (closure: DialogResponseClosure, values?: IDictionary) => void;
}

export interface IConfirmDialogState extends IComponentState {
    title?: string;
    message: React.ReactNode;
    buttons: IConfirmDialogButtons;
    values?: IDictionary;
    description?: string[];
}

export class ConfirmDialog extends Component<IConfirmDialogProperties, IConfirmDialogState> {

    private dialogRef = React.createRef<Dialog>();

    public constructor(props: IConfirmDialogProperties) {
        super(props);
        this.state = {
            message: "",
            buttons: {},
        };
        this.addHandledProperties("onClose");
    }

    public show = (message: React.ReactNode, buttons: IConfirmDialogButtons, title?: string, description?: string[],
        values?: IDictionary): void => {
        this.setState({ title, message, buttons, values, description }, () => {
            return this.dialogRef.current?.open({ closeOnEscape: true });
        });
    };

    public render(): React.ReactNode {
        const { title, message, buttons, description } = this.state;

        const className = this.getEffectiveClassNames(["confirmDialog"]);
        let dialogContent = null;
        if (React.isValidElement(message)) {
            dialogContent = message;
        } else {
            // If no explicit content is specified, use the description list for additional content.
            const descriptionLabels: React.ReactNode[] = [];
            description?.forEach((value, index) => {
                descriptionLabels.push(
                    <Label
                        id={`caption${index}`}
                        language="text"
                        caption={value}
                    />,
                );
            });

            dialogContent =
                <Container orientation={Orientation.TopDown}>
                    {message && <Label id="dialogMessage" caption={message as string} />}
                    <Container
                        orientation={Orientation.TopDown}
                        className="description">
                        {descriptionLabels}
                    </Container>
                </Container>;
        }

        // TODO: consider the different order of the buttons based on the OS.
        const actions: React.ReactNode[] = [];
        if (buttons.alternative) {
            actions.push(<Button
                caption={buttons.alternative.replace(/&/g, "")} // Remove hot key indicator. We cannot show them.
                id="alternative"
                key="alternative"
                isDefault={buttons.alternative === buttons.default}
                onClick={this.handleActionClick}
            />);
        }

        if (buttons.accept) {
            actions.push(<Button
                caption={buttons.accept.replace(/&/g, "")}
                id="accept"
                key="accept"
                isDefault={buttons.accept === buttons.default}
                onClick={this.handleActionClick}
            />);
        }

        if (buttons.refuse) {
            actions.push(<Button
                caption={buttons.refuse.replace(/&/g, "")}
                id="refuse"
                key="refuse"
                isDefault={buttons.refuse === buttons.default}
                onClick={this.handleActionClick}
            />);
        }

        return (
            <Dialog
                ref={this.dialogRef}
                className={className}
                caption={
                    <>
                        <Icon src={Codicon.Question} />
                        <Label>{title ?? "Confirm"}</Label>
                    </>
                }
                content={dialogContent}
                actions={{
                    end: actions,
                }}
                onClose={this.handleClose}
            >
            </Dialog>
        );
    }

    private handleActionClick = (e: React.SyntheticEvent, props: Readonly<IComponentProperties>): void => {
        const { onClose } = this.props;
        const { values } = this.state;

        let closure;
        switch (props.id) {
            case "accept": {
                closure = DialogResponseClosure.Accept;
                break;
            }

            case "alternative": {
                closure = DialogResponseClosure.Alternative;
                break;
            }

            default: {
                closure = DialogResponseClosure.Decline;
                break;
            }
        }

        this.dialogRef.current?.close(false);

        onClose?.(closure, values);
    };

    private handleClose = (cancelled: boolean): void => {
        if (cancelled) {
            const { onClose } = this.mergedProps;
            const { values } = this.state;

            onClose?.(DialogResponseClosure.Cancel, values);
        }
    };

}
