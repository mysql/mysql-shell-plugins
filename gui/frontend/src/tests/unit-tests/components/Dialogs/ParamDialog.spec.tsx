/*
 * Copyright (c) 2022, Oracle and/or its affiliates.
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License, version 2.0,
 * as published by the Free Software Foundation.
 *
 * This program is also distributed with certain software (including
 * but not limited to OpenSSL) that is licensed under separate terms, as
 * designated in a particular file or component or in included license
 * documentation.  The authors of MySQL hereby grant you an additional
 * permission to link the program and your derivative works with the
 * separately licensed software that they have included with MySQL.
 * This program is distributed in the hope that it will be useful,  but
 * WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See
 * the GNU General Public License, version 2.0, for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program; if not, write to the Free Software Foundation, Inc.,
 * 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA
 */

import React from "react";
import { mount } from "enzyme";
import keyboardKey from "keyboard-key";

import { changeInputValue, nextProcessTick, sendKeyPress, snapshotFromWrapper } from "../../test-helpers";
import { ParamDialog } from "../../../../components/Dialogs/ParamDialog";

describe("Param Dialog Tests", (): void => {
    it("Render Test", () => {
        const component = mount<ParamDialog>(
            <ParamDialog
                caption="Lorem Ipsum"
            />,
        );

        expect(snapshotFromWrapper(component)).toMatchSnapshot();
        component.unmount();
    });

    it("Show Panel and Escape", async () => {
        const spyOnClose = jest.fn((_accepted: boolean, _payload?: unknown): void => {
            // no-op
        });

        let portals = document.getElementsByClassName("portal");
        expect(portals.length).toBe(0);

        const component = mount<ParamDialog>(
            <ParamDialog
                caption="Dolor Sit Amet"
                onClose={spyOnClose}
            />,
        );

        component.instance().show();
        await nextProcessTick();

        portals = document.getElementsByClassName("portal");
        expect(portals.length).toBe(1);
        expect(portals[0]).toMatchSnapshot();

        sendKeyPress(keyboardKey.Escape);
        await nextProcessTick();

        portals = document.getElementsByClassName("portal");
        expect(portals.length).toBe(0);
        expect(spyOnClose).toHaveBeenCalledTimes(1);
        expect(spyOnClose).toHaveBeenCalledWith(true, { name: "", value: "" });

        component.unmount();
    });

    it("Show Panel and Escape", async () => {
        const spyOnClose = jest.fn((_accepted: boolean, _payload?: unknown): void => {
            // no-op
        });

        let portals = document.getElementsByClassName("portal");
        expect(portals.length).toBe(0);

        const component = mount<ParamDialog>(
            <ParamDialog
                onClose={spyOnClose}
            />,
        );

        component.instance().show();
        await nextProcessTick();

        portals = document.getElementsByClassName("portal");
        expect(portals.length).toBe(1);
        expect(portals[0]).toMatchSnapshot();
        expect(component.state()).toStrictEqual({ name: "", value: "" });

        const inputs = portals[0].getElementsByClassName("input");
        expect(inputs).toHaveLength(2);
        expect(inputs[0].id).toBe("name");
        expect(inputs[1].id).toBe("value");

        changeInputValue(inputs[0], "ABC");
        await nextProcessTick();
        expect(component.state().name).toBe("ABC");

        changeInputValue(inputs[1], "XYZ");
        await nextProcessTick();
        expect(component.state().value).toBe("XYZ");

        const buttons = portals[0].getElementsByClassName("button");
        expect(buttons).toHaveLength(3);
        expect(buttons[1].id).toBe("ok");
        expect(buttons[2].id).toBe("cancel");

        (buttons[1] as HTMLButtonElement).click();
        expect(spyOnClose).toHaveBeenCalledTimes(1);
        expect(spyOnClose).toHaveBeenCalledWith(false, { name: "ABC", value: "XYZ" });
        await nextProcessTick();

        portals = document.getElementsByClassName("portal");
        expect(portals.length).toBe(0);

        component.unmount();
    });
});
