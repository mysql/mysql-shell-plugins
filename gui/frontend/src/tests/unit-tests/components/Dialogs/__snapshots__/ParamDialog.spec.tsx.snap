// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Param Dialog Tests Render Test 1`] = `
<ParamDialog
  caption="Lorem Ipsum"
>
  <Dialog
    actions={
      Object {
        "end": Array [
          <Button
            caption="OK"
            id="ok"
            onClick={[Function]}
          />,
          <Button
            caption="Cancel"
            id="cancel"
            onClick={[Function]}
          />,
        ],
      }
    }
    caption={
      <y>
        <Icon
          disabled={false}
          src={255}
        />
        <Label>
          Parameters
        </Label>
      </y>
    }
    className="msg paramDialog"
    container={Object {}}
    content={
      <Grid
        columnGap={8}
        columns={
          Array [
            "128px",
            "auto",
            "auto",
          ]
        }
      >
        <GridCell
          rowSpan={4}
        >
          <Icon
            disabled={false}
            height={128}
            src="parameters.svg"
            width={128}
          />
        </GridCell>
        <GridCell
          columnSpan={2}
          crossAlignment="center"
        >
          <Label
            caption="Lorem Ipsum"
            id="caption"
          />
        </GridCell>
        <GridCell
          className="left"
          crossAlignment="center"
        >
          Parameter name:
        </GridCell>
        <GridCell
          className="right"
          crossAlignment="center"
        >
          <Input
            id="name"
            onChange={[Function]}
            spellCheck={true}
            value=""
          />
        </GridCell>
        <GridCell
          className="left"
          crossAlignment="center"
        >
          Parameter value:
        </GridCell>
        <GridCell
          className="right"
          crossAlignment="center"
        >
          <Input
            id="value"
            onChange={[Function]}
            spellCheck={true}
            value=""
          />
        </GridCell>
      </Grid>
    }
    onClose={[Function]}
  >
    <Portal
      container={Object {}}
      onClose={[Function]}
      onOpen={[Function]}
    />
  </Dialog>
</ParamDialog>
`;

exports[`Param Dialog Tests Show Panel and Escape 1`] = `
<div
  __self="[object Object]"
  __source="[object Object]"
  class="msg portal"
>
  <div
    __self="[object Object]"
    __source="[object Object]"
    class="msg dialog visible paramDialog"
  >
    <div
      __self="[object Object]"
      __source="[object Object]"
      class="title verticalCenterContent"
    >
      <div
        __self="[object Object]"
        __source="[object Object]"
        class="msg icon codicon codicon-pass-filled"
      />
      <label
        __self="[object Object]"
        __source="[object Object]"
        class="msg label"
        data-tooltip="expand"
      >
        Parameters
      </label>
      <button
        __self="[object Object]"
        __source="[object Object]"
        class="msg button imageOnly"
        id="closeButton"
        tabindex="0"
      >
        <div
          __self="[object Object]"
          __source="[object Object]"
          class="msg icon"
          style="mask-image: url(close2.svg);"
        />
      </button>
    </div>
    <div
      __self="[object Object]"
      __source="[object Object]"
      class="content fixedScrollbar"
    >
      <div
        __self="[object Object]"
        __source="[object Object]"
        class="msg grid"
        style="grid-template-columns: 128px auto auto; column-gap: 8px;"
      >
        <div
          __self="[object Object]"
          __source="[object Object]"
          class="msg container fixedScrollbar gridCell"
          style="flex-direction: row; grid-row: span 4;"
        >
          <div
            __self="[object Object]"
            __source="[object Object]"
            class="msg icon"
            style="mask-image: url(parameters.svg); width: 128px; height: 128px;"
          />
        </div>
        <div
          __self="[object Object]"
          __source="[object Object]"
          class="msg container fixedScrollbar gridCell"
          style="flex-direction: row; align-items: center; grid-column: span 2;"
        >
          <label
            __self="[object Object]"
            __source="[object Object]"
            class="msg label"
            data-tooltip="expand"
            id="caption"
          >
            Dolor Sit Amet
          </label>
        </div>
        <div
          __self="[object Object]"
          __source="[object Object]"
          class="msg container fixedScrollbar gridCell left"
          style="flex-direction: row; align-items: center;"
        >
          Parameter name:
        </div>
        <div
          __self="[object Object]"
          __source="[object Object]"
          class="msg container fixedScrollbar gridCell right"
          style="flex-direction: row; align-items: center;"
        >
          <input
            __self="[object Object]"
            __source="[object Object]"
            class="msg input"
            id="name"
            spellcheck="true"
            type="text"
          />
        </div>
        <div
          __self="[object Object]"
          __source="[object Object]"
          class="msg container fixedScrollbar gridCell left"
          style="flex-direction: row; align-items: center;"
        >
          Parameter value:
        </div>
        <div
          __self="[object Object]"
          __source="[object Object]"
          class="msg container fixedScrollbar gridCell right"
          style="flex-direction: row; align-items: center;"
        >
          <input
            __self="[object Object]"
            __source="[object Object]"
            class="msg input"
            id="value"
            spellcheck="true"
            type="text"
          />
        </div>
      </div>
    </div>
    <div
      __self="[object Object]"
      __source="[object Object]"
      class="footer verticalCenterContent"
    >
      <div
        __self="[object Object]"
        __source="[object Object]"
        class="msg container fixedScrollbar leftItems"
        style="flex-direction: row;"
      />
      <div
        __self="[object Object]"
        __source="[object Object]"
        class="msg container fixedScrollbar rightItems"
        style="flex-direction: row-reverse;"
      >
        <button
          __self="[object Object]"
          __source="[object Object]"
          caption="OK"
          class="msg button"
          id="ok"
          tabindex="0"
        >
          OK
        </button>
        <button
          __self="[object Object]"
          __source="[object Object]"
          caption="Cancel"
          class="msg button"
          id="cancel"
          tabindex="0"
        >
          Cancel
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Param Dialog Tests Show Panel and Escape 2`] = `
<div
  __self="[object Object]"
  __source="[object Object]"
  class="msg portal"
>
  <div
    __self="[object Object]"
    __source="[object Object]"
    class="msg dialog visible paramDialog"
  >
    <div
      __self="[object Object]"
      __source="[object Object]"
      class="title verticalCenterContent"
    >
      <div
        __self="[object Object]"
        __source="[object Object]"
        class="msg icon codicon codicon-pass-filled"
      />
      <label
        __self="[object Object]"
        __source="[object Object]"
        class="msg label"
        data-tooltip="expand"
      >
        Parameters
      </label>
      <button
        __self="[object Object]"
        __source="[object Object]"
        class="msg button imageOnly"
        id="closeButton"
        tabindex="0"
      >
        <div
          __self="[object Object]"
          __source="[object Object]"
          class="msg icon"
          style="mask-image: url(close2.svg);"
        />
      </button>
    </div>
    <div
      __self="[object Object]"
      __source="[object Object]"
      class="content fixedScrollbar"
    >
      <div
        __self="[object Object]"
        __source="[object Object]"
        class="msg grid"
        style="grid-template-columns: 128px auto auto; column-gap: 8px;"
      >
        <div
          __self="[object Object]"
          __source="[object Object]"
          class="msg container fixedScrollbar gridCell"
          style="flex-direction: row; grid-row: span 4;"
        >
          <div
            __self="[object Object]"
            __source="[object Object]"
            class="msg icon"
            style="mask-image: url(parameters.svg); width: 128px; height: 128px;"
          />
        </div>
        <div
          __self="[object Object]"
          __source="[object Object]"
          class="msg container fixedScrollbar gridCell"
          style="flex-direction: row; align-items: center; grid-column: span 2;"
        >
          <label
            __self="[object Object]"
            __source="[object Object]"
            class="msg label"
            data-tooltip="expand"
            id="caption"
          >
            Please enter name/vale pair:
          </label>
        </div>
        <div
          __self="[object Object]"
          __source="[object Object]"
          class="msg container fixedScrollbar gridCell left"
          style="flex-direction: row; align-items: center;"
        >
          Parameter name:
        </div>
        <div
          __self="[object Object]"
          __source="[object Object]"
          class="msg container fixedScrollbar gridCell right"
          style="flex-direction: row; align-items: center;"
        >
          <input
            __self="[object Object]"
            __source="[object Object]"
            class="msg input"
            id="name"
            spellcheck="true"
            type="text"
          />
        </div>
        <div
          __self="[object Object]"
          __source="[object Object]"
          class="msg container fixedScrollbar gridCell left"
          style="flex-direction: row; align-items: center;"
        >
          Parameter value:
        </div>
        <div
          __self="[object Object]"
          __source="[object Object]"
          class="msg container fixedScrollbar gridCell right"
          style="flex-direction: row; align-items: center;"
        >
          <input
            __self="[object Object]"
            __source="[object Object]"
            class="msg input"
            id="value"
            spellcheck="true"
            type="text"
          />
        </div>
      </div>
    </div>
    <div
      __self="[object Object]"
      __source="[object Object]"
      class="footer verticalCenterContent"
    >
      <div
        __self="[object Object]"
        __source="[object Object]"
        class="msg container fixedScrollbar leftItems"
        style="flex-direction: row;"
      />
      <div
        __self="[object Object]"
        __source="[object Object]"
        class="msg container fixedScrollbar rightItems"
        style="flex-direction: row-reverse;"
      >
        <button
          __self="[object Object]"
          __source="[object Object]"
          caption="OK"
          class="msg button"
          id="ok"
          tabindex="0"
        >
          OK
        </button>
        <button
          __self="[object Object]"
          __source="[object Object]"
          caption="Cancel"
          class="msg button"
          id="cancel"
          tabindex="0"
        >
          Cancel
        </button>
      </div>
    </div>
  </div>
</div>
`;
